File: Frontend/src/app/models/Processor.ts
import {Part} from "./Part";
import {Socket} from "./Socket";

export interface Processor {
    id: number;
    part: Part;
    socket: Socket;
    cores: number;
    threads: number;
    clockFreq: number;
    turboFreq: number;
    oc: boolean;
    graph: boolean;
}

File: Backend/Domain/Processor.cs
using System;

namespace Domain
{
    public class Processor
    {
        public int Id { get; set; }
        public Part Part { get; set; }
        public Socket Socket { get; set; }
        public int Cores { get; set; }
        public int Threads { get; set; }
        public int ClockFreq { get; set; }
        public int TurboFreq { get; set; }
        public bool Oc { get; set; }
        public bool Graph { get; set; }
    }
}